sql/src/main/java/io/crate/planner/operators/Rename.java

```diff
@@ -80,6 +84,41 @@ public RelationName name() {
         return outputs;
     }
 
+    @Override
+    public LogicalPlan pruneOutputsExcept(Collection<Symbol> outputsToKeep) {
+        /* In `SELECT * FROM (SELECT t1.*, t2.* FROM tbl AS t1, tbl AS t2) AS tjoin`
+         * The `ScopedSymbol`s are ambiguous; To map them correctly this uses a IdentityHashMap
+         */
+        IdentityHashMap<Symbol, Symbol> parentToChildMap = new IdentityHashMap<>(outputs.size());
```

Comment:
I'm not too happy about having to resort to identity here, but I don't really have a better idea.